### PR TITLE
API-8597 add validation and save internal api info

### PR DESCRIPTION
**Pull request by michaelkimball**
[API-8597](https://vajira.max.gov/browse/API-8597)

Add Joi validation to request with tests
Save information into DynamoDB
Send VA email to Slack﻿

### CODE REVIEW

File: routes/DeveloperApplication.ts [hunk 1]
```diff
@@ -10,15 +10,21 @@ import OktaService from '../services/OktaService';
...
 
 export const applySchema = Joi.object()
   .keys({
     apis: Joi.custom(validateApiList).required(),
     description: Joi.string().allow(''),
     email: Joi.string().email().required(),
     firstName: Joi.string().required(),
+    internalApiInfo: Joi.object().keys({
+      programName: Joi.string().required(),
+      sponsorEmail: Joi.string().email().required(),
```
Comment (christineyost) [inline routes/DeveloperApplication.ts:25]: Does sponsorEmail not need to be a `va.gov` email?

File: routes/DeveloperApplication.test.ts [hunk 2]
```diff
@@ -440,6 +440,86 @@ describe('validations', () => {
...
+        const internalApiInfo = { ...defaultInternalApiInfo, vaEmail: 12345 };
+        const payload = { ...defaultPayload, internalApiInfo };
+
+        const result = applySchema.validate(payload);
+
+        expect(result.error?.message).toEqual('"internalApiInfo.vaEmail" must be a string');
+      });
+
+      it('is in a valid format', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, vaEmail: 'lolnotanemail.com' };
```
Comment (christineyost) [inline routes/DeveloperApplication.test.ts:511]: would it also be worth testing a valid email that doesn't end with `va.gov`?

File: models/User.test.ts [hunk 3]
```diff
@@ -57,6 +60,16 @@ describe('User', () => {
...
+
+    it('errors when signing up for internal api without va email', () => {
+      expect.assertions(1);
+      event.apis = 'addressValidation';
+      try {
+        user = new User(event);
+      } catch (err: unknown) {
+        expect((err as Error).message).toEqual('Applying for internal api without VA email');
+      }
+    });
```
Comment (christineyost) [inline models/User.test.ts:72]: would it be worth adding tests for if there is a vaEmail but it doesn't end with `va.gov` as well as positive tests for the `email` and `vaEmail` properties containing valid VA emials?

File: routes/DeveloperApplication.ts (hunk 1 above)
Comment (michaelkimball) [reply routes/DeveloperApplication.ts:25]: I don't think so, but I'm not sure tbh

File: routes/DeveloperApplication.test.ts (hunk 2 above)
Comment (michaelkimball) [reply routes/DeveloperApplication.test.ts:511]: Good call out

File: models/User.test.ts (hunk 3 above)
Comment (michaelkimball) [reply models/User.test.ts:72]: Will add

File: routes/DeveloperApplication.ts (hunk 1 above)
Comment (michaelkimball) [reply routes/DeveloperApplication.ts]: Confirmed that `va.gov` is required on sponsorEmail. Added the relevant validation and tests.

**Review by christineyost — APPROVED** [review]
Looks good to me! 🎉

**Pull request merged**